src/bridges/chainlink-ccip/CcipGhoBridge.sol

```diff
@@ -0,0 +1,177 @@
+
+    if (totalAmount == 0) {
+      revert InvalidTransferAmount();
+    }
+
+    IERC20(GHO).transferFrom(msg.sender, address(this), totalAmount);
+    IERC20(GHO).approve(ROUTER, totalAmount);
+
+    Client.EVMTokenAmount[] memory tokenAmounts = new Client.EVMTokenAmount[](1);
+    tokenAmounts[0] = Client.EVMTokenAmount({token: GHO, amount: totalAmount});
+
+    Client.EVM2AnyMessage memory message = Client.EVM2AnyMessage({
+      receiver: abi.encode(bridges[destinationChainSelector]),
+      data: abi.encode(transfers),
+      tokenAmounts: tokenAmounts,
+      extraArgs: '',
+      feeToken: payFeesIn == PayFeesIn.LINK ? LINK : address(0)
+    });
+
+    uint256 fee = IRouterClient(ROUTER).getFee(destinationChainSelector, message);
+
+    if (payFeesIn == PayFeesIn.LINK) {
+      LinkTokenInterface(LINK).transferFrom(msg.sender, address(this), fee);
+      LinkTokenInterface(LINK).approve(ROUTER, fee);
+      messageId = IRouterClient(ROUTER).ccipSend(destinationChainSelector, message);
+    } else {
+      if (msg.value < fee) {
+        revert InsufficientFee();
+      }
+
+      messageId = IRouterClient(ROUTER).ccipSend{value: fee}(destinationChainSelector, message);
+      if (msg.value > fee) {
+        payable(msg.sender).transfer(msg.value - fee);
+      }
+    }
+
+    emit TransferIssued(messageId, destinationChainSelector, totalAmount);
+  }
+
+  /// @inheritdoc ICcipGhoBridge
+  function quoteTransfer(
+    uint64 destinationChainSelector,
+    Transfer[] calldata transfers,
+    PayFeesIn payFeesIn
+  ) external view checkDestination(destinationChainSelector) returns (uint256 fee) {
+    uint256 totalAmount;
+
+    for (uint256 i; i < transfers.length; ) {
+      totalAmount += transfers[i].amount;
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    if (totalAmount == 0) {
+      revert InvalidTransferAmount();
+    }
+
+    Client.EVMTokenAmount[] memory tokenAmounts = new Client.EVMTokenAmount[](1);
+    tokenAmounts[0] = Client.EVMTokenAmount({token: GHO, amount: totalAmount});
+
+    Client.EVM2AnyMessage memory message = Client.EVM2AnyMessage({
+      receiver: abi.encode(bridges[destinationChainSelector]),
+      data: abi.encode(transfers),
+      tokenAmounts: tokenAmounts,
+      extraArgs: '',
+      feeToken: payFeesIn == PayFeesIn.LINK ? LINK : address(0)
+    });
+
+    fee = IRouterClient(ROUTER).getFee(destinationChainSelector, message);
+  }
+
+  /// @inheritdoc CCIPReceiver
+  function _ccipReceive(Client.Any2EVMMessage memory message) internal override {
+    bytes32 messageId = message.messageId;
+    uint64 sourceChainSelector = message.sourceChainSelector;
+    address bridge = abi.decode(message.sender, (address));
+    Transfer[] memory transfers = abi.decode(message.data, (Transfer[]));
+
+    if (bridge != bridges[sourceChainSelector]) {
+      revert InvalidMessage();
+    }
+
+    for (uint256 i; i < transfers.length; ) {
+      IERC20(GHO).transfer(transfers[i].to, transfers[i].amount);
```

Comment:
We likely only want to transfer to the collector

Reply:
But I think generalization is good for the future

Reply:
@LucasWongC Probably won't be approved by BGD